src/components/enketo-iframe.vue

```diff
@@ -102,11 +107,26 @@ function handleIframeMessage(event) {
     let eventData;
     try { eventData = JSON.parse(event.data); } catch {}
 
-    if (eventData?.enketoEvent === 'submissionsuccess' && redirectUrl.value) {
-      router.push((new URL(redirectUrl.value)).pathname);
+    if (eventData?.enketoEvent === 'submissionsuccess') {
+      if (props.actionType === 'edit') {
+        // for edit we always redirect to Submission details page
+        router.push(submissionPath(form.projectId, form.xmlFormId, props.instanceId));
+      } else if (!props.actionType && redirectUrl.value) {
+        // for public link, we read return value from query parameter. The value could be 3rd party
+        // site as well, typically a thank you page
+        try {
+          const normalizedUrl = new URL(redirectUrl.value);
+          if (normalizedUrl.origin === location.origin) {
+            router.push(normalizedUrl.pathname);
+          } else {
+            location.assign(normalizedUrl);
+          }
+        } catch (e) {}
```

Comment:
How about adding a test of the `catch` case? Basically just checking that if the specified URL is invalid, the user stays on the page.